lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1691,25 +1686,44 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder hacb) {
         }
         return restClient;
     }
+
+    public Response makeRequest(
+            final String reqType,
+            final String urlPath,
+            final Map<String, ?> doc) 
+            throws IOException {
+        return makeRequest(reqType, urlPath, doc, Collections.emptyMap());
+    }
     
-    public Response makeBulkRequest(String reqType, String indexName, File jsonData) 
+    public Response makeRequestBulk(
+            final String reqType,
+            final String indexName,
+            final File jsonData) 
             throws IOException {
-        RestClient restClient = getRestClient();
         try (InputStream is = new FileInputStream(jsonData)) {
-            InputStreamEntity body = new InputStreamEntity(is);
-            Response response = restClient.performRequest(reqType, "/" + indexName + "/_bulk", 
-                    Collections.emptyMap(), body);
-            return response;
+            return makeRequest(reqType, "/" + indexName + "/_bulk", Collections.emptyMap(),
+                    new InputStreamEntity(is));
         }
     }
     
-    public Response makeRequest(String reqType, String urlPath, Map<String, ?> doc, 
-            Map<String, String> attributes) throws IOException {
+    private Response makeRequest(
+            final String reqType,
+            final String urlPath,
+            final Map<String, ?> doc, 
+            final Map<String, String> attributes)
+            throws IOException {
+        return makeRequest(reqType, urlPath, attributes, doc == null ? null : stringEntity(
```

Comment:
(HttpEntity)stringEntity(
                UObject.transformObjectToString(doc))

Reply:
Why is the cast necessary?

Reply:
Helps the reader match the arguments to the right function signature.

Reply:
So you're advocating for including a cast every time a polymorphic method is called and the type is unclear?

Reply:
My mistake. My IDE was not making the connection to the function signature and it was an error with the way I manually included the jars. Please ignore.